### DISCUSSION TITLE
How to get CPU and memory usage of my application?

### DISCUSSION THREAD

**Discussion by LahariReddyMuthyala** [open-telemetry/opentelemetry-cpp#2671] [Q&A]
Hello, is there a way for me to get CPU and memory usage of my application in C++? I see an equivalent for this in Java. 
If this is not supported, can I have some suggestions on how I have achieve this with minimal performance impact? 

Thank you for your help!

**Comment by lalitb** [comment]
Do you want to capture the CPU and memory usage of your application, and send it as metrics to some backend. Can you share what is there for Java?

**Reply by LahariReddyMuthyala** [reply]
Yes. I want to capture the CPU and memory usage as metric and send it in Prometheus format. I found [this](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/ProcessMetrics.java#L19) for Java.

**Reply by lalitb** [reply]
Ok thanks. We don't have any such instrumentation in main or contrib repo.  I did create a draft PR some time back for the support on POSIX systems, but didn't had time to complete it. If you are interested to explore more - https://github.com/open-telemetry/opentelemetry-cpp-contrib/pull/218

**Reply by LahariReddyMuthyala** [reply]
Thank you for your response @lalitb! Is it possible to generate the metrics using other libraries (open to suggestions on libraries, if you have any in mind) and export it using OpenTelemetry?

**Reply by lalitb** [reply]
You can try finding some cross-platform C++ library (sorry I am not aware of any such) to collect the CPU and memory stats, and then feed these stats to OpenTelemetry using the [Metrics API](https://github.com/open-telemetry/opentelemetry-cpp/tree/main/api/include/opentelemetry/metrics). This definitely won't be straightforward, so I was trying to create one specifically for OpenTelemetry. Sorry I know this is not much of help :(

**Reply by LahariReddyMuthyala** [reply]
No worries! Thank you so much for your inputs.
